lib/3scale/backend/worker.rb

```diff
@@ -30,7 +26,15 @@ def self.new(options = {})
           WorkerMetrics.start_metrics_server
         end
 
-        super
+        if options[:async]
+          # Conditional require is done to require async-* libs only when
+          # needed and avoid possible side-effects.
```

Comment:
Looks good - must ensure the same happens when loading the async storage in tests, or in general in the listeners when choosing just the sync one (I say this because I haven't checked it, but you probably know).